File: 30-Day-Leetcoding-Challenge/02-Happy-Number/Happy-Number.cpp
class Solution {
public:
    int get_next(int n){
        int sum = 0;
        while(n){
            sum += (n % 10) * (n % 10);
            n /= 10;
        }
        return sum;
    }
    bool isHappy(int n) {
        set<int> vis;
        while(n != 1){
            vis.insert(n);
            n = get_next(n);
            if (vis.find(n) != vis.end())
                return false;
        }
        return true;
    }
};

File: 30-Day-Leetcoding-Challenge/02-Happy-Number/Happy-Number.cs
public class Solution
{
    public bool IsHappy(int n)
    {
        var hs = new HashSet<int>();
        var left = 0;
        //HashSet is the key
        while (hs.Add(n))
        {
            var sum = 0;
            while (n > 0)
            {
                left = n % 10;
                sum += left * left;
                n /= 10;
            }

            if (sum == 1)
            {
                return true;
            }
            else
            {
                n = sum;
            }
        }
        return false;
    }
}